src/basics.md

```diff
@@ -411,6 +412,45 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-simple-error-handling]: #ex-simple-error-handling
+<a name="ex-simple-error-handling"></a>
+## Handle errors correctly in main
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Handles error that occur when we try to open a file that does not exist. We do this using [error-chain], a library that takes care of a lot of boilerplate code needed in order to [handle errors in Rust].
+
+We call `Io(std::io::Error)` inside `foreign_links` to convert the type `std::io::Error` into error-chain's custom version of it.
+
+To handle the result of a function that might return an error, in this example `open_foo`, it must return a `Result`. To handle that result, we use pattern matching on it when calling the function. In this example we print an error message if that result was an error.
+
+
+
+```rust
+#[macro_use]
+extern crate error_chain;
+
+error_chain!{
+    foreign_links {
+        Io(std::io::Error);
+    }
+}
+
+fn main() {
+    if let Err(ref e) = read_foo() {
+        // handle the error here
+        println!("error: {}", e);
+    }
+}
+
+fn read_foo() -> Result<()> {
+    use std::fs::File;
+    File::open("foo.txt")?;
+
```

Comment:
Very nice :+1: 

You can skip the `foo` variable, and there is no need to have `String` in parens. Also the use statements would be on top level.
```rust
use std::fs::File;
use std::io::{Result, Read};

fn read_foo() -> Result<String> {
    let mut foo_content = String::new();
    File::open("foo.txt")?.read_to_string(&mut foo_content)?;
    Ok(foo_content)
}

fn main() {
    match read_foo() {
        Ok(foo_content) => println!("{}", foo_content),
        Err(error) => println!("error: {}", error),
    };
}
```

Unfortunately this example can be done fully without any need for error_chain (as presented above)

So we might need to think on a relatively non contrived way to introduce another error variant except  `Io(std::io::Error);` to justify error_chain usage :smile_cat:. You can either:
- define your own app error and return it if foo_content does not match some arbitrary predicate like (starts_with("#"))
- get the filename from env var (and we already have another error variant)
- parse the foo_contents into something like an int
- pass the path by value and manipulate it (for instance strip)
- your idea